File: Raven.Abstractions/Data/StreamResult.cs
// -----------------------------------------------------------------------
//  <copyright file="StreamResult.cs" company="Hibernating Rhinos LTD">
//      Copyright (c) Hibernating Rhinos LTD. All rights reserved.
//  </copyright>
// -----------------------------------------------------------------------
using Raven.Json.Linq;

namespace Raven.Abstractions.Data
{
	public class StreamResult<T>
	{
		 public string Key { get; set; }
		 public Etag Etag { get; set; }
		 public RavenJObject Metadata { get; set; }
		 public T Document { get; set; }
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/abstractions/data/StreamResult.java
package net.ravendb.abstractions.data;

import net.ravendb.abstractions.json.linq.RavenJObject;

public class StreamResult<T> {
  private String key;
  private Etag etag;
  private RavenJObject metadata;
  private T document;

  public String getKey() {
    return key;
  }
  public void setKey(String key) {
    this.key = key;
  }
  public Etag getEtag() {
    return etag;
  }
  public void setEtag(Etag etag) {
    this.etag = etag;
  }
  public RavenJObject getMetadata() {
    return metadata;
  }
  public void setMetadata(RavenJObject metadata) {
    this.metadata = metadata;
  }
  public T getDocument() {
    return document;
  }
  public void setDocument(T document) {
    this.document = document;
  }


}
